### PR TITLE
mon, osdmap: optimize check and cleanup of pg_temp for new osdmaps

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/osd/OSDMap.cc [hunk 1]
```diff
@@ -1175,81 +1175,80 @@ void OSDMap::dedup(const OSDMap *o, OSDMap *n)
...
+      }
+    }
+    if (num_up == 0) {
+      ldout(cct, 10) << __func__ << "  removing pg_temp " << p->first
+		     << " with all down osds" << p->second << dendl;
+      pending_inc->new_pg_temp[p->first].clear();
+      continue;
+    }
+    // redundant pg_temp?
+    if (pending_inc->new_pg_temp.count(p->first) == 0) {
```
Comment (xiexingguo) [inline src/osd/OSDMap.cc]: I think it should be 
```
if (pending_inc->new_pg_temp.count(p->first)) 
```
here.

**Comment by xiexingguo** [comment]
Otherwise looks good.

File: src/osd/OSDMap.cc (hunk 1 above)
Comment (liewegas) [inline src/osd/OSDMap.cc]: Hrm.. actually this an dthe other check are both sort of wrong.  Will revise.

**Comment by liewegas** [comment]
@xiexingguo updated, please look again?  Thanks!

File: src/osd/OSDMap.cc [hunk 2]
```diff
@@ -1235,15 +1235,20 @@ void OSDMap::remove_down_temps(CephContext *cct,
...
 	++num_up;
+	break;
+      }
     }
-    if (num_up == 0)
+    if (num_up == 0) {
+      ldout(cct, 10) << __func__ << "  removing pg_temp " << p->first
+		     << " with all down osds" << p->second << dendl;
       pending_inc->new_pg_temp[p->first].clear();
+      continue;
```
Comment (tchaikov) [inline src/osd/OSDMap.cc]: nit, could remove `continue` here.

**Comment by xiexingguo** [comment]
looks good to me now, sir:-)

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2016-05-13_16:00:42-rados-wip-yuri-testing-distro-basic-smithi/

**Pull request merged**